honeybee_3dm/togeometry.py

```diff
@@ -1,34 +1,311 @@
+    lb_faces, colors = extract_mesh_faces_colors(mesh, color_by_face)
+    return Mesh3D(lb_verts, lb_faces, colors)
+
+
+def brep_to_face3d(brep):
+    """Create a Ladybug Face3D object from a rhino3dm Brep.
+
+    Args:
+        brep: A rhino3dm Brep.
+
+    Returns:
+        A list of Ladybug Face3D objects.
+    """
+    faces = []
+    for i in range(len(brep.Faces)):
+        mesh = brep.Faces[i].GetMesh(rhino3dm.MeshType.Any)
+        faces.extend(mesh_to_face3d(mesh))
+
+    return faces
+
+
+def brep2d_to_face3d(brep, tolerance):
+    """Create a Ladybug Face3D object from a rhino3dm Brep.
+
+    This function is used in translating rhino3dm planar Brep to Ladybug Face3D objects.
+
+    Args:
+        brep: A rhino3dm Brep.
+
+    Returns:
+        A list of Ladybug Face3D objects.
+    """
+    faces = []
+    # * Check 01 - If any of the edge is an arc
+    lines = []
+    not_line = 0
+    for i in range(len(brep.Edges)):
+        if brep.Edges[i].IsLinear(tolerance) is False:
+            not_line += 1
+    # If there's an arc in one of the edges, mesh it
+    if not_line > 0:
+        warnings.warn(
+            "The surface has curved edges. It will be meshed.")
+        faces.extend(brep_to_face3d(brep))
+        return faces
+    else:
+        # Create Ladybug lines from start and end points of edges
+        for i in range(len(brep.Edges)):
+            start_pt = to_point3d(brep.Edges[i].PointAtStart)
+            end_pt = to_point3d(brep.Edges[i].PointAtEnd)
+            line = LineSegment3D.from_end_points(start_pt, end_pt)
+            lines.append(line)
+
+        # Create Ladybug Polylines from the lines
+        polylines = Polyline3D.join_segments(lines, tolerance)
+        # Sort the Ladybug Polylines based on length
+        # The longest polyline belongs to the boundary of the face
+        # The rest of the polylines belong to the holes in the face
+        sorted_polylines = sorted(polylines, key=lambda polyline: polyline.length)
+        # The first polyline in the list shall always be the polyline for
+        # the boundary
+        sorted_polylines.reverse()
+
+        # Getting all vertices of the face
+        mesh = brep.Faces[0].GetMesh(rhino3dm.MeshType.Any)
+        try:
+            mesh_pts = [to_point3d(mesh.Vertices[i])
+                        for i in range(len(mesh.Vertices))]
+        except AttributeError:
+            warnings.warn(
+                'Non-planar Breps are not handled by Honeybee. They are ignored.'
+            )
+            return faces
+        else:
+            # In the list of the Polylines, if there's only one polyline then
+            # the face has no holes
+            if len(sorted_polylines) == 1:
+                if len(mesh_pts) == 4 or len(mesh_pts) == 3:
+                    faces.extend(mesh_to_face3d(mesh))
+                else:
+                    boundary_pts = remove_dup_vertices(
+                        sorted_polylines[0].vertices, tolerance)
```

Comment:
Why do you need to do this here?  The `Face3D` method has it's own method for removing duplicate vertices which people can optionally do afterwards if its needed for a simulation.   For example, Radiance doesn't care about duplicate vertices but EnergyPlus does (so all of the to_idf methods and to_osm methods remove them). Also, you are missing the case here where a 4-vertex triangular face has a duplicate vertex.  So it's not like your code here is actually reliable anyway.

Reply:
The vertices extracted from mesh have duplicates and unless I remove them I cannot create a Face3D. That's why I had to create that helper function to remove duplicate vertices.